### DISCUSSION TITLE
Protractor Inheritance from ElementFinder class

### DISCUSSION THREAD

**Discussion by NavyMint** [webdriverio/codemod#41] [Migration Issue] [answered]
Hello guys. We are in a process of migration to Wdio. And stuck on a wrappers classes inherited from protractor ElementFinder. Is there a way to implement same way wrappers in wdio?
For exmpl 
class BaseElementWrapper extends Webdriver.Element 

P.S: I've tried to inherit from Webdriver.Element but no luck, what is the best way to do this?

**Comment by christian-bromann** [comment, marked as answer]
You can use `addCommand` to add custom commands to the WebdriverIO element object. See more details on that [in the docs](https://webdriver.io/docs/customcommands). Then you can extend the TypeScript definition by doing:

```ts
declare global {
    namespace WebdriverIO {
        interface Element {
            elementCustomCommand: (arg: unknown) => number
        }
    }
}
```

**Reply by NavyMint** [reply]
thanks, Christian

**Reply by hoancmg** [reply]
@NavyMint  Could you please show the wrapper class? I'm doing this to. Thanks in advance!

**Reply by NavyMint** [reply]
@hoancmg hey, sorry I can't share real class due to project policy, but if you have questions I can try to answer.
Overall we are using general approach simmilar to this one https://github.com/Xotabu4/protractor-element-extend#creating-own-single-fragment
But moving all this from protractor to wdio. In wdio class looks like here simple exmpl at the top
https://github.com/erwinheitzman/wdio-typings/blob/master/test/specs/example.e2e.ts
